controller/controller.go

```diff
@@ -220,14 +220,38 @@ func (c *Controller) RunOnce(ctx context.Context) error {
 		ManagedRecords:     c.ManagedRecordTypes,
 	}
 
-	plan = plan.Calculate()
+	p = p.Calculate()
 
-	if plan.Changes.HasChanges() {
-		err = c.Registry.ApplyChanges(ctx, plan.Changes)
+	if p.Changes.HasChanges() {
+		err = c.Registry.ApplyChanges(ctx, p.Changes)
+		// If the original batch fails, try each action separately
+		// And if the single action batch fails, try batch halves
 		if err != nil {
-			registryErrorsTotal.Inc()
-			deprecatedRegistryErrors.Inc()
-			return err
+			create, update, delete := splitChanges(p.Changes)
+
+			createErr := c.Registry.ApplyChanges(ctx, create)
+			if createErr != nil {
+				err = applyHalfActionChanges(c, ctx, create)
```

Comment:
To keep this PR small and get some experience with the general approach: how about we separate the PR into two and only to the "create,update,delete"-split here. It makes the PR much easier (and also less useful) but it would already solve something and we can get experience with the approach taken here.

Otherwise, I'd recommend to also finish the "half-apply" functionality because currently it's only splitting once with doesn't solve our problem when we have broken records scattered throughout the batch.